coramin/third_party/tests/test_minlplib_tools.py

```diff
@@ -0,0 +1,143 @@
+import coramin
+import unittest
+import os
+
+
+class TestMINLPLibTools(unittest.TestCase):
+    def test_get_minlplib_instancedata(self):
+        current_dir = os.path.abspath(os.path.dirname(__file__))
+        coramin.third_party.get_minlplib_instancedata(download_dir=os.path.join(current_dir, 'minlplib'))
+        self.assertTrue(os.path.exists(os.path.join(current_dir, 'minlplib', 'instancedata.csv')))
+        os.remove(os.path.join(current_dir, 'minlplib', 'instancedata.csv'))
+        os.rmdir(os.path.join(current_dir, 'minlplib'))
```

Comment:
Why are you doing a download as part of a test?  That means you will get test failures when you run tests without an internet connection.  I would recommend doing the download outside of the test suite proper.

Reply:
Wouldn't you get test failures either way if you didn't have an internet connection (you wouldn't be able to download the files)? I could instead skip the tests if minlplib.org cannot be reached. What do you think?

Reply:
I added a check at the top of the test file to skip the tests if minlplib.org cannot be reached. Let me know what you think.

Reply:
FWIW, what Pyomo does is to set up code coverage so that it is automatically activated for all python processes (that is also how we get subprocess coverage).  Then we run scripts to download the extensions as part of the github workflow action.  If the downloads fail then we fail the build.  The tests then run, either assuming that the extensions are present or skipping tests if the extension(s) aren't present.

The relevant chunks of code are [here](https://github.com/Pyomo/pyomo/blob/e49f63eba52043d27db4a33e61187bf84987ff07/.github/workflows/mpi_matrix_test.yml#L97-L122).  Note that the coverage configuration includes
```
[run]
parallel = True
```
and you need to run `coverage combine` before uploading the coverage results.

The key point is that the downloads are part of the *build process* and not the *test harness*.